documents/src/pages/utils/navigation-grid.md

```diff
@@ -5,8 +5,443 @@ type: page
+
+const grid = [
+  [1, 0, 1, 0],
+  [1, 1, 1, 1],
+  [1, 1, 0, 1],
+  [0, 1, 1, 1]
+];
+
+last(grid); // => [3, 3]
+```
+
+#### Syntax
+
+```text
+last(grid);
+```
+
+#### Arguments
+
+| Name | Type | Description |
+| --- | --- | --- |
+| grid | NavigationGrid | The navigation grid |
+
+#### Returns
+
+| Type | Description |
+| --- | --- |
+| CellIndex &#124; null | The last active cell index |
+
+### left
+
+Navigate _left_ from the cell. If there is no active cell on the _left_, then iterate over preceding rows to find one.
+
+```typescript
+import { left } from '@refinitiv-ui/utils/navigation.js';
+
+const grid = [
+  [1, 0, 1, 0],
+  [1, 1, 1, 1],
+  [1, 1, 0, 1],
+  [0, 1, 1, 1]
+];
+
+left(grid, [1, 2]); // => [0, 2]
+left(grid, [0, 1]); // => [2, 0]
+left(grid, [0, 0]); // => null
+```
+
+#### Syntax
+
+```text
+left(grid, cell);
+```
+
+#### Arguments
+
+| Name | Type | Description |
+| --- | --- | --- |
+| grid | NavigationGrid | The navigation grid |
+| cell | CellIndex | The cell index |
+
+#### Returns
+
+| Type | Description |
+| --- | --- |
+| CellIndex &#124; null | The previous active cell index |
+
+### right
+
+Navigate _right_ from the cell. If there is no active cell on the _right_, then iterate over following rows to find one.
+
+```typescript
+import { right } from '@refinitiv-ui/utils/navigation.js';
+
+const grid = [
+  [1, 0, 1, 0],
+  [1, 1, 1, 1],
+  [1, 1, 0, 1],
+  [0, 1, 1, 1]
+];
+
+right(grid, [1, 2]); // => [3, 2]
+right(grid, [3, 2]); // => [1, 3]
+right(grid, [3, 3]); // => null
+```
+
+#### Syntax
+
+```text
+right(grid, cell);
+```
+
+#### Arguments
+
+| Name | Type | Description |
+| --- | --- | --- |
+| grid | NavigationGrid | The navigation grid |
+| cell | CellIndex | The cell index |
+
+#### Returns
+
+| Type | Description |
+| --- | --- |
+| CellIndex &#124; null | The next active cell index |
+
+### up
+
+Navigate _up_ from the cell to find the closest active cell on the preceding rows.
+
+```typescript
+import { up } from '@refinitiv-ui/utils/navigation.js';
+
+const grid = [
+  [1, 0, 1, 0],
+  [1, 1, 1, 1],
+  [1, 1, 0, 1],
+  [0, 1, 1, 1]
+];
+
+up(grid, [1, 2]); // => [1, 1]
+up(grid, [1, 1]); // => [2, 0]
+up(grid, [0, 0]); // => null
+```
+
+#### Syntax
+
+```text
+up(grid, cell);
+```
+
+#### Arguments
+
+| Name | Type | Description |
+| --- | --- | --- |
+| grid | NavigationGrid | The navigation grid |
+| cell | CellIndex | The cell index |
+
+#### Returns
+
+| Type | Description |
+| --- | --- |
+| CellIndex &#124; null | The closest active cell index on the preceding rows |
+
+## Types
+
+The list of _TypeScript_ types.
+
+| Name | Description | Value |
+| --- | --- | --- |
+| CellIndex | Cell index as `[Column Index, Row Index]`. _index_ is 0-based. | `[number, number]` |
+| NavigationGrid | A list of `NavigationRow` | `NavigationRow[]` |
+| NavigationRow | A list of _0_ and _1_. _0_ is an inactive cell, but _1_ is active. | `(0 &#124; 1)[]` |
```

Comment:
```suggestion
| NavigationRow | A list of _0_ and _1_. _0_ is an inactive cell, but _1_ is active. | `(0 | 1)[]` |
```
I think we can use "|", when it wrap with backticks.

Reply:
maybe, my IDE built-in editor does not like it